--- animeface.c.bak     2016-08-01 04:02:45.974197643 +0000
+++ animeface.c 2016-08-01 04:05:07.674279793 +0000
@@ -20,14 +20,15 @@
 nv_conv_imager2nv(nv_matrix_t *bgr, nv_matrix_t *gray,
                                  VALUE im)
 {
-       int y, x,  xsize, ysize, depth;
+       int y, x,  xsize, ysize, range, depth;
        VALUE c;
        ID rb_id_green, rb_id_blue, rb_id_red, rb_id_pixel_color;

        xsize = NUM2INT(rb_funcall(im, rb_intern("columns"), 0));
        ysize = NUM2INT(rb_funcall(im, rb_intern("rows"), 0));
-       depth = NUM2INT(rb_const_get(cMagick, rb_intern("QuantumDepth")));
-
+       range = NUM2INT(rb_const_get(cMagick, rb_intern("QuantumRange")));
+       depth = log10((range + 1)) / log10(2.0);
+
        assert(bgr->rows == gray->rows && bgr->rows == ysize);
        assert(bgr->cols == gray->cols && bgr->cols == xsize);
